src/device.coffee

```diff
@@ -128,13 +128,11 @@ exports.getDeviceType = do ->
 					throw new Error('Device type not specified in config file')
 				return configFromFile.deviceType
 
-# Calling this function updates the local device state, which is then used to synchronise
-# the remote device state, repeating any failed updates until successfully synchronised.
-# This function will also optimise updates by merging multiple updates and only sending the latest state.
-exports.updateState = do ->
+do ->
 	applyPromise = Promise.resolve()
 	targetState = {}
 	actualState = {}
+	updatePending = false
```

Comment:
Not exactly - that one is tri-state and refers to the application.update function itself (works as a sort of self-mutex). This one explicitly gives a true/false response to whether there's an update that should be applied but the supervisor can't apply it, specifically what users want when using this together with the update lock.

Moreover, we can't require application.coffee here since we'd have circular requires.